metron-analytics/metron-maas-common/src/main/java/org/apache/metron/maas/discovery/ServiceDiscoverer.java

```diff
@@ -76,12 +92,19 @@ public void resetState() {
 
   private void updateState() {
     Map<Model, List<ModelEndpoint>> state = new HashMap<>();
+    Map<String, String> modelToVersion = new HashMap<>();
     Map<String, ServiceInstance<ModelEndpoint>> containerToEndpoint = new HashMap<>();
     try {
       for(String name : serviceDiscovery.queryForNames()) {
         for(ServiceInstance<ModelEndpoint> endpoint: serviceDiscovery.queryForInstances(name)) {
           ModelEndpoint ep = endpoint.getPayload();
           LOG.info("Found model endpoint " + ep);
+          String currentVersion = modelToVersion.getOrDefault(ep.getName(), ep.getVersion());
+          modelToVersion.put( ep.getName()
+                            , currentVersion.compareTo(ep.getVersion()) < 0
+                            ? ep.getVersion()
+                            : currentVersion
+                            );
```

Comment:
I am not sure that I totally understand what the intention here is.  I've had to stare at this code block for like 10 minutes.  Is there no way to make this an easier read?  Just my opinion; maybe I'm slow.